com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -312,6 +310,40 @@ define([
                         }
                     }
                 }
+
+                if(form_id == 'end_date' && !$scope.edit_data[role_id].end_date){
+                    $scope.edit_data[role_id].end_date = null;
+                }
+
+                if ($scope.edit_data[role_id].job_contract_id
+                    && $scope.edit_data[role_id].start_date
+                    && typeof $scope.edit_data[role_id].end_date != 'undefined'
+                    && (form_id == 'start_date' || form_id == 'job_contract_id' || form_id == 'end_date')) {
+
+                    $scope.updateRolesWithContractData(role_id);
+                }
+            };
+
+            $scope.updateRolesWithContractData = function (role_id) {
+                var contract_id = $scope.edit_data[role_id].job_contract_id;
```

Comment:
Please use camelCase (`contractId`) instead of snake_case (`contract_id`) when possible. Some of the property names come directly from the backend so they're stuck with the snake_case, but other than those the standard is the camelCase